sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategy.scala

```diff
@@ -57,9 +57,10 @@ import org.apache.spark.sql.types._
 private[sql] object FileSourceStrategy extends Strategy with Logging {
   def apply(plan: LogicalPlan): Seq[SparkPlan] = plan match {
     case PhysicalOperation(projects, filters, l@LogicalRelation(files: HadoopFsRelation, _, _))
-      if (files.fileFormat.toString == "TestFileFormat" ||
-         files.fileFormat.isInstanceOf[parquet.DefaultSource]) &&
-         files.sqlContext.conf.parquetFileScan =>
+      if files.fileFormat.toString == "TestFileFormat" ||
+         (files.fileFormat.isInstanceOf[parquet.DefaultSource] &&
+         files.sqlContext.conf.parquetFileScan) ||
+         files.fileFormat.isInstanceOf[text.DefaultSource] =>
```

Comment:
No need to be addressed in this PR, but we should probably have a more general way to do this kind of data source dispatching. For example, ORC is in the hive package and can't be directly referenced here.